src/typecheck/mod.rs

```diff
@@ -1023,6 +1029,31 @@ pub fn apparent_type(t: &Term, envs: Option<&Envs>) -> ApparentType {
     }
 }
 
+/// Infer type of a more complex structure.
+/// For now, it's implemented only to infer the type of a record on wich fields are typed.
+/// It's use esencialy to type the stdlib, but could be use in a future for types placeholders for
+/// instence.
```

Comment:
It won't be used for placeholders, though. We will use full-fledged, standard type inference for those. So it's really just for the stdlib.